src/common/hash/index.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { ChildProcess, fork } from 'child_process';
+import { join } from 'path';
+import { HashRequest, HashResponse } from './hash';
+import { debounce } from '../objUtils';
+
+let instance: ChildProcess | undefined;
+let messageId = 0;
+
+const cleanup = debounce(30 * 1000, () => {
+  instance?.kill();
+  instance = undefined;
+});
+
+const create = () => {
+  if (instance) {
+    return instance;
+  }
+
+  instance = fork(join(__dirname, 'hash.js'), [], { env: {}, silent: true });
+  instance.setMaxListeners(Infinity);
+  return instance;
+};
+
+const send = (req: HashRequest): Promise<string | undefined> => {
+  const cp = create();
+  cleanup();
```

Comment:
Does this call just start the 30s timer for the cleanup callback? It's not totally clear from the name alone.

Reply:
Yep, `debouce` will call the wrapped function 30 seconds after calls to it stop. [More explanation on debouncing](https://medium.com/@jamischarles/what-is-debouncing-2505c0648ff1)

Reply:
Got it. I'm familiar with debouncing, it just wasn't clear that calling `cleanup()` wasn't actually executing the cleanup method, but rather _starting the debounce timer for the cleanup method_.

Reply:
Ah, gotcha, I'll clarify that variable name on `master`!